src/actions/vstudio/vs2010_nuget.lua

```diff
@@ -54,33 +37,148 @@
 		end
 	end
 
+	function nuget2010.packageAPIInfo(prj, package)
+		local id = nuget2010.packageId(package)
+		local version = nuget2010.packageVersion(package)
 
---
--- Generates the packages.config file.
---
+		if not packageAPIInfos[package] then
+			local packageAPIInfo = {}
 
-	function nuget2010.generatePackagesConfig(obj)
-		local wks = obj.workspace
+			printf("Examining NuGet package '%s'...", id)
+			io.flush()
 
-		local done = {}
-		local packages = {}
-		for prj in p.workspace.eachproject(wks) do
-			for i = 1, #prj.nuget do
-				local package = prj.nuget[i]
+			local response, err, code = http.get(string.format("https://api.nuget.org/v3/registration1/%s/index.json", id:lower()))
```

Comment:
I just tried this with MyGet, and while it looks like while they implement the same API as `api.nuget.org`, they don't return a file listing for the packages (at least if they were uploaded by their automatic `packages.config` tool). There's something weird going on here.

`api.nuget.org` doesn't return a file listing for all packages either, but from my testing that only seemed to happen with older packages.

If the API doesn't provide us with a file listing, the only other way that I can think of is to have Premake download and extract the packages.